kedro/io/partitioned_data_set.py

```diff
@@ -53,6 +54,13 @@
 S3_PROTOCOLS = ("s3", "s3a", "s3n")
 
 
+def _grandparent(path: str) -> str:
+    path_obj = Path(path)
```

Comment:
We should probably either use `pathlib.PurePosixPath` here, or `return grandparent.as_posix()` below (or both) to make sure we don't end up returning Windows paths with `\` separator for S3, for example.

Reply:
Ah! You're probably right. I had `path_obj = Path(path).resolve()` before which messed it up for S3-ish, so I took it out, but this must be why Windows is failing.

Reply:
Sorry for the delay on this. Seems it fixed the Windows test errors!